### ISSUE TITLE
Pipeline depends on packages with bad k8s imports, preventing minor upgrades

### ISSUE THREAD

**Issue by fejta** [tektoncd/pipeline#1244]
See https://github.com/kubernetes/kubernetes/issues/81878 and/or https://github.com/google/go-containerregistry/issues/496

The reconciler imports go-containerregistry's k8schain package, which has an unsupported k8s.io/kubernetes import.

```console
$ go mod why k8s.io/kubernetes/pkg/credentialprovider/secrets

github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1
github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1.test
github.com/tektoncd/pipeline/test/builder
github.com/tektoncd/pipeline/pkg/reconciler/v1alpha1/taskrun/resources
github.com/tektoncd/pipeline/pkg/reconciler/v1alpha1/taskrun/entrypoint
github.com/google/go-containerregistry/pkg/authn/k8schain
k8s.io/kubernetes/pkg/credentialprovider/secrets
```

This makes it impossible for downstream pipeline consumers to do minor module upgrades:

```console
$ go get -u k8s.io/test-infra # or anything else which imports tektoncd/pipeline
...
go: k8s.io/cloud-provider@v0.0.0: unknown revision v0.0.0
go: k8s.io/metrics@v0.0.0: unknown revision v0.0.0
go: k8s.io/kube-proxy@v0.0.0: unknown revision v0.0.0
go: k8s.io/apiserver@v0.0.0: unknown revision v0.0.0
go: k8s.io/kube-scheduler@v0.0.0: unknown revision v0.0.0
go: k8s.io/csi-translation-lib@v0.0.0: unknown revision v0.0.0
go: k8s.io/cri-api@v0.0.0: unknown revision v0.0.0
go: k8s.io/sample-apiserver@v0.0.0: unknown revision v0.0.0
go: k8s.io/code-generator@v0.0.0: unknown revision v0.0.0
go get: error loading module requirements
```

**Comment by vdemeester** [comment]
/assign
I think the problem should show up for recent version of tektoncd/pipeline… I'll try to bump the dependency in `kubernetes/test-infra`

**Comment by vdemeester** [comment]
Hum interesting, `go mod` doesn't really look into what package is used and what is not… So even if `test-infra` doesn't import anything in tektoncd/pipeline that uses `go-containerregistry`, it lists it as an indirect dependency…

**Comment by vdemeester** [comment]
Quick update, https://github.com/google/go-containerregistry/pull/649 might be the temporary fix (waiting for https://github.com/kubernetes/enhancements/pull/1406) to remove that import path.

**Comment by vdemeester** [comment]
@fejta I think depending on latest pipeline once 0.10 is released should fix this :angel:

**Comment by vdemeester** [comment]
https://github.com/kubernetes/test-infra/pull/16099 is merged, 0.10 is released, We can close :tada: 
/close
